lib/msprime.c

```diff
@@ -1589,6 +1589,86 @@ msp_defrag_segment_chain(msp_t *self, segment_t *z)
 }
 
 static int WARN_UNUSED
+msp_recombine(msp_t *self, segment_t *x, segment_t **u, segment_t **v)
+{
+    int ret = 0;
+    int ix;
+    double mu;
+    uint64_t k;
+    segment_t *y, *z;
+    segment_t s1, s2;
+    segment_t *seg_tails[] = {&s1, &s2};
+
+    if ( self->recombination_rate > 0 ) {
+        mu = 1.0 / self->recombination_rate;
+        k = 1 + (uint64_t) x->left +
+            (uint64_t) gsl_ran_exponential(self->rng, mu);
+    } else {
+        mu = INFINITY;
+        k = INT64_MAX;
```

Comment:
This should probably be UINT64_MAX, since k is a uint64_t. I'm not clear on how this interacts with the code below. Would you mind putting in a quick comment here describing what happens in the zero recombination case. 

Come to think of it, why don't we just skip the function entirely when rho = 0?